### DISCUSSION TITLE
Get own IP address

### DISCUSSION THREAD

**Discussion by mglazzari-qinmotion** [cesanta/mongoose#2869] [Ask questions, get answers] [answered]
Is there a way for the main program to obtain it own IP address of a server running in Mongoose? 
If I set the listening address to 0.0.0.0 so Mongoose obtains an IP address from a DHCP server, how can I know what the IP address is, so that I can, for instance, show it on a display? 
In case it is relevant, this is an STM32H7 platform.

**Comment by scaprile** [comment, marked as answer]
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
https://github.com/cesanta/mongoose/blob/97296934ab594241ba2489f018a21779882a7936/examples/stm32/nucleo-h723zg-make-baremetal-builtin/main.c#L84-L93

If using auto-init code (Wizard):
https://github.com/cesanta/mongoose/blob/97296934ab594241ba2489f018a21779882a7936/src/drivers/stm32h.h#L39-L53
then redefine macro above as you wish or cast `mgr->priv->ip` appropriately. e.g.:
https://github.com/cesanta/mongoose/blob/97296934ab594241ba2489f018a21779882a7936/src/net_builtin.c#L1124-L1125
https://github.com/cesanta/mongoose/blob/97296934ab594241ba2489f018a21779882a7936/src/net_builtin.c#L888-L900

**Comment by mglazzari-qinmotion** [comment]
OK, so I added this:

```c
uint32_t mongoose_get_ip(){
	struct mg_tcpip_if *ifp = (struct mg_tcpip_if *) g_mgr.priv;
	return ifp->ip;
}
```
in mongoose_impl.c (one of the files generated by the wizard), and it works. Thanks!
